lib/BasicEvaluatedExpression.js

```diff
@@ -2,156 +2,180 @@
-BasicEvaluatedExpression.prototype.isConstArray = function() {
-	return Object.prototype.hasOwnProperty.call(this, "array");
-};
-BasicEvaluatedExpression.prototype.isIdentifier = function() {
-	return Object.prototype.hasOwnProperty.call(this, "identifier");
-};
-BasicEvaluatedExpression.prototype.isWrapped = function() {
-	return Object.prototype.hasOwnProperty.call(this, "prefix") || Object.prototype.hasOwnProperty.call(this, "postfix");
-};
-BasicEvaluatedExpression.prototype.isTemplateString = function() {
-	return Object.prototype.hasOwnProperty.call(this, "quasis");
-}
-BasicEvaluatedExpression.prototype.asBool = function() {
-	if(this.isBoolean()) return this.bool;
-	else if(this.isNull()) return false;
-	else if(this.isString()) return !!this.string;
-	else if(this.isNumber()) return !!this.number;
-	else if(this.isRegExp()) return true;
-	else if(this.isArray()) return true;
-	else if(this.isConstArray()) return true;
-	else if(this.isWrapped()) return this.prefix && this.prefix.asBool() || this.postfix && this.postfix.asBool() ? true : undefined;
-	else if(this.isTemplateString()) {
-		if(this.quasis.length === 1) return this.quasis[0].asBool();
-		for(var i = 0; i < this.quasis.length; i++) {
-			if(this.quasis[i].asBool()) return true;
+"use strict";
+
+class BasicEvaluatedExpression {
+
+	constructor() {
+		this.range = null;
+	}
+
+	isNull() {
+		return !!this.null;
+	}
+
+	isString() {
+		return Object.prototype.hasOwnProperty.call(this, "string");
+	}
+
+	isNumber() {
+		return Object.prototype.hasOwnProperty.call(this, "number");
+	}
+
+	isBoolean() {
+		return Object.prototype.hasOwnProperty.call(this, "bool");
+	}
+
+	isRegExp() {
+		return Object.prototype.hasOwnProperty.call(this, "regExp");
+	}
+
+	isConditional() {
+		return Object.prototype.hasOwnProperty.call(this, "options");
+	}
+
+	isArray() {
+		return Object.prototype.hasOwnProperty.call(this, "items");
+	}
+
+	isConstArray() {
+		return Object.prototype.hasOwnProperty.call(this, "array");
+	}
+
+	isIdentifier() {
+		return Object.prototype.hasOwnProperty.call(this, "identifier");
+	}
+
+	isWrapped() {
+		return Object.prototype.hasOwnProperty.call(this, "prefix") || Object.prototype.hasOwnProperty.call(this, "postfix");
+	}
+
+	isTemplateString() {
+		return Object.prototype.hasOwnProperty.call(this, "quasis");
+	}
+
+	asBool() {
+		if(this.isBoolean()) return this.bool;
+		else if(this.isNull()) return false;
+		else if(this.isString()) return !!this.string;
+		else if(this.isNumber()) return !!this.number;
+		else if(this.isRegExp()) return true;
+		else if(this.isArray()) return true;
+		else if(this.isConstArray()) return true;
+		else if(this.isWrapped()) return this.prefix && this.prefix.asBool() || this.postfix && this.postfix.asBool() ? true : undefined;
+		else if(this.isTemplateString()) {
+			if(this.quasis.length === 1) return this.quasis[0].asBool();
+			for(var i = 0; i < this.quasis.length; i++) {
+				if(this.quasis[i].asBool()) return true;
+			}
```

Comment:
Actually, the logic here is wrong; ``` `${0}` ``` should be truthy. But that's a different problem, and a bug in the original 😅